### PR TITLE
RC4: Converting to text template

### PR DESCRIPTION
**Pull request by mintsoft**
###### Description of changes

Ref: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2767

---

Instant Answer Page: https://duck.co/ia/view/rc4

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @puskin94

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/Rc4.t [hunk 1]
```diff
@@ -9,22 +9,25 @@ use DDG::Test::Goodie;
...
-		result => "grYU1K8="
-	}),
-	'rc4 de duck yWrJniG/nNg=' => test_zci("RC4 Decrypt: yWrJniG/nNg=, with key: duck is DdgRocks",
-	structured_answer => {
-		input => ['yWrJniG/nNg=, Key: duck'],
-		operation => "RC4 Decrypt",
-		result => "DdgRocks"
-	}),
+    'rc4 en mysecretkey hello' => build_test("RC4 Encrypt: hello, with key: mysecretkey is grYU1K8=", "grYU1K8=", "RC4 Encrypt: hello, Key: mysecretkey"),
+	'rc4 de duck yWrJniG/nNg=' => build_test("RC4 Decrypt: yWrJniG/nNg=, with key: duck is DdgRocks", "DdgRocks", "RC4 Decrypt: yWrJniG/nNg=, Key: duck"),
```
Comment (GuiltyDolphin) [inline t/Rc4.t]: We could just pass in `(decrypt/encrypt), text, key`

File: lib/DDG/Goodie/Rc4.pm [hunk 2]
```diff
@@ -43,11 +43,14 @@ handle remainder => sub {
...
 
-    return "$operation: $plaintext, with key: $key is $result",
-    structured_answer => {
-        operation => $operation,
-        input => [html_enc($plaintext) . ", Key: $key"],
-        result => $result
+    return "$operation: $plaintext, with key: $key is $result", structured_answer => {
+        data => {
+            title => html_enc($result),
+            subtitle => "$operation: " . html_enc($plaintext) . ", Key: $key"
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Rc4.pm]: We might have to html_enc the `$key` too.

**Comment by GuiltyDolphin** [comment]
@mintsoft Still a couple of things to do on this one!

**Comment by mintsoft** [comment]
@GuiltyDolphin All done

**Comment by moollaza** [comment]
Deploying to beta...

File: t/Rc4.t [hunk 3]
```diff
@@ -9,22 +9,25 @@ use DDG::Test::Goodie;
...
-		result => "grYU1K8="
-	}),
-	'rc4 de duck yWrJniG/nNg=' => test_zci("RC4 Decrypt: yWrJniG/nNg=, with key: duck is DdgRocks",
-	structured_answer => {
-		input => ['yWrJniG/nNg=, Key: duck'],
-		operation => "RC4 Decrypt",
-		result => "DdgRocks"
-	}),
+    'rc4 en mysecretkey hello' => build_test("Encrypt", "hello", "mysecretkey", "grYU1K8="),
+	'rc4 de duck yWrJniG/nNg=' => build_test("Decrypt", "yWrJniG/nNg=", "duck", "DdgRocks"),
```
Comment (GuiltyDolphin) [inline t/Rc4.t]: Indent seems to have gone awry!

**Comment by GuiltyDolphin** [comment]
@mintsoft Thanks! :dolphin:

**Pull request merged**